timdex_dataset_api/dataset.py

```diff
@@ -469,3 +389,82 @@ def log_write_statistics(self, start_time: float) -> None:
             f"total rows: {total_rows}, "
             f"total size: {total_size}"
         )
+
+    def read_batches_iter(
```

Comment:
Commenting here but applies to all the `read_*` methods: None of the `read_*` methods actually update `self.dataset`. Does this mean that it is always and only the `load` method that updates `self.dataset`?

I think this is an important distinction to note somewhere in future documentation (if not already documented somewhere beyond our PR discussions). 🤔

Reply:
That's correct.  Once the dataset is loaded, by design, any filtering performed in the read methods do not modify the loaded dataset.  As you mention above, I think this will prove helpful in TIM where you could:

1. load the dataset with `run_date` and `run_id`
2. perform a read, _additioanlly_ filtering by `action="index"`
3. perform another read, this time filtering by `action="delete"`

Both reads work, as they don't modify the originally loaded dataset.  That's what [this part of a test](https://github.com/MITLibraries/timdex-dataset-api/pull/52/files#diff-b2a37482b2c59ca7f4496a3205006af83b1a741ffef35367941f60a0fdec818cR59-R60) is meant to exercise.